### PR TITLE
fix for defect 402 - batch reference

### PR DESCRIPTION
**Pull request by trevstephenson**
Relationship deletes were not being setup correctly when batching.  This is a fix for defect https://github.com/simple-odata-client/Simple.OData.Client/issues/402.

### CODE REVIEW

File: src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs [hunk 1]
```diff
@@ -134,9 +134,10 @@ public async Task<ODataRequest> CreateLinkRequestAsync(string collection, string
         public async Task<ODataRequest> CreateUnlinkRequestAsync(string collection, string linkName, string entryIdent, string linkIdent)
         {
             var associationName = _session.Metadata.GetNavigationPropertyExactName(collection, linkName);
-            await WriteEntryContentAsync(RestVerbs.Delete, collection, entryIdent, null, false).ConfigureAwait(false);
-
             var commandText = FormatLinkPath(entryIdent, associationName, linkIdent);
+            
+            await WriteEntryContentAsync(RestVerbs.Delete, collection, commandText, null, false).ConfigureAwait(false);
```
Comment (watfordgnf) [inline src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs]: Can you add a comment in the code why this ordering is important?

Comment (watfordgnf) [reply src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs]: Also a test case showing that this re-ordering resolves the problem you describe.

Comment (trevstephenson) [reply src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs]: If you look at the signature of the WriteEntryContentAsync method being called, you can see it expects the commandText variable in the place of the third paramater. In the code as it was before I modified it, the commandText variable is being populated after the call to the WriteEntryContentAsync method was being called. It looked like a typo to me, so I moved the line below and corrected the parameter being passed, then stepped it through and it looked ok. Also inspected the calls in Fiddler and they look correct.

I'm hoping someone else has time to construct a test case as I'm a little pressed for time. However, to reproduce the issue you can have a look at the bug 402. Basically you just need do an unlink (https://github.com/simple-odata-client/Simple.OData.Client/wiki/Updating-entries-with-links#update-a-product-and-unlink-a-category) in a Batch UpdateEntryAsync (https://github.com/simple-odata-client/Simple.OData.Client/wiki/Batch-requests) and you will see this problem.

I've used this fix a lot with Microsoft PowerApps/Dynamics 365 Web API and it works like a charm.

Comment (trevstephenson) [inline src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs]: f you look at the signature of the WriteEntryContentAsync method being called, you can see it expects the commandText variable in the place of the third paramater. In the code as it was before I modified it, the commandText variable is being populated after the call to the WriteEntryContentAsync method was being called. It looked like a typo to me, so I moved the line below and corrected the parameter being passed, then stepped it through and it looked ok. Also inspected the calls in Fiddler and they look correct.

I'm hoping someone else has time to construct a test case as I'm a little pressed for time. However, to reproduce the issue you can have a look at the bug #402. Basically you just need do an unlink (https://github.com/simple-odata-client/Simple.OData.Client/wiki/Updating-entries-with-links#update-a-product-and-unlink-a-category) in a Batch UpdateEntryAsync (https://github.com/simple-odata-client/Simple.OData.Client/wiki/Batch-requests) and you will see this problem.

I've used this fix a lot with Microsoft PowerApps/Dynamics 365 Web API and it works like a charm.

**Comment by object** [comment]
Sorry for long delay. PR is tested and merged, thanks a lot!

**Pull request merged**